header/verify.go

```diff
@@ -23,43 +38,39 @@ func (eh *ExtendedHeader) Verify(untrusted libhead.Header) error {
 	}
 
 	isAdjacent := eh.Height()+1 == untrst.Height()
-	if isAdjacent {
-		// Optimized verification for adjacent headers
-		// Check the validator hashes are the same
-		if !bytes.Equal(untrst.ValidatorsHash, eh.NextValidatorsHash) {
-			return &libhead.VerifyError{
-				Reason: fmt.Errorf("expected old header next validators (%X) to match those from new header (%X)",
-					eh.NextValidatorsHash,
-					untrst.ValidatorsHash,
-				),
-			}
-		}
-
-		if !bytes.Equal(untrst.LastHeader(), eh.Hash()) {
-			return &libhead.VerifyError{
-				Reason: fmt.Errorf("expected new header to point to last header hash (%X), but got %X)",
-					eh.Hash(),
-					untrst.LastHeader(),
-				),
-			}
-		}
-
+	if !isAdjacent {
 		return nil
 	}
```

Comment:
Does that mean every header will be just accepted as fully valid now? Or will be verified with a adjacent one at a later point? If so, there should be a comment here explaining why simply returning here is fine.

Reply:
Only non-adjacent headers will skip adjacent header verification logic, this is okay for sync targets - but we will always verify the chain sequentially moving forward.

Reply:
done.